pages/hub/apps/guides/advanced/live-tutorial/data-retrieval.mdx

```diff
@@ -0,0 +1,313 @@
+
+If you've utilized fields excluded from the Live subgraph, we recommend splitting data fetching into `games` and `conditions`.
+This allows you to reuse game/sport queries for both Prematch and Live requests.
+
+Here's an example of a universal hook to retrieve events from both Prematch and Live feeds:
+
+<Tabs items={[ 'useSportsAndEvents.ts', 'query.ts' ]}>
+<Tabs.Tab>
+```ts {6,11,17-19,22-23,29} copy
+import { universalGamesQuery } from './query'
+
+const useSportsAndEvents = (props) => {
+  const {
+    startsAt,
+    isLive,
+    ...yourRestVariablesLikeSportSlug,
+  } = props
+
+  const { prematchClient, liveClient } = useApolloClients()
+  const client = isLive ? liveClient! : prematchClient!
+
+  const variables = {
+    ...yourRestVariablesLikeSportSlug,
+  }
+
+  if (isLive) {
+    // LIVE: _lt suffix
+    variables.where.startsAt_lt = startsAt
+  }
+  else {
+    // PREMATCH: _gt suffix + LP filter
+    variables.where.startsAt_gt = startsAt
+    variables.where.liquidityPool = _lp_contract_address.toLowerCase()
+  }
+
+  return useQuery(universalGamesQuery, {
+    variables,
+    client,
+  })
+}
+```
+</Tabs.Tab>
+<Tabs.Tab>
+```ts copy
+export const universalGamesQuery = gql`
+  query Games(
+    $first: Int,
+    $skip: Int,
+    $where: Game_filter!,
+    $orderBy: Game_orderBy,
+    $orderDirection: OrderDirection,
+  ) {
+    games(first: $first, skip: $skip, where: $where, orderBy: $orderBy, orderDirection: $orderDirection, subgraphError: allow) {
+      id
+      gameId
+      title
+      startsAt
+      status
+      sport {
+        sportId
+        slug
+        name
+      }
+      league {
+        slug
+        name
+        country {
+          slug
+          name
+        }
+      }
+      participants {
+        image
+        name
+      }
+    }
+  }
+`;
+```
+</Tabs.Tab>
+</Tabs>
+
+Apply this condition tweak to every fetching event data query.
+
+<Callout type="advice">
+  Check out the SDK implementation of event fetching hooks for both Prematch and Live feeds:
+
+  - Sports navigation: https://github.com/Azuro-protocol/sdk/blob/v3.1.0/src/hooks/useSportsNavigation.ts
+  - Events: https://github.com/Azuro-protocol/sdk/blob/v3.1.0/src/hooks/useSports.ts
+  - Games: https://github.com/Azuro-protocol/sdk/blob/v3.1.0/src/hooks/useGames.ts
+  - Game: https://github.com/Azuro-protocol/sdk/blob/v3.1.0/src/hooks/useGame.ts
+</Callout>
+
+## Load markets
+
+Because of the difference between Live and Prematch `condition` entities,
+it's not possible to reuse the Prematch query. You'll need to create a new one.
+
+Take a look at the source code of the SDK's [useGameMarkets](https://github.com/Azuro-protocol/sdk/blob/v3.1.0/src/hooks/useGameMarkets.ts) hook.
+It fetches conditions from the Live feed when the game is live, and Prematch conditions when the game hasn't started yet.
+To streamline the handling of both prematch and live conditions in the SDK, we're modifying these entities to have the same fields.
```

Comment:
actually useConditions does that, useGameMarkets combines conditions in markets

Reply:
https://github.com/Azuro-protocol/gem-docs/pull/155/commits/f9e8ff61f1172594f6046f4b2fbbfab8f0422e1f